### DISCUSSION TITLE
Posts deletion feature

### DISCUSSION THREAD

**Discussion by PavanLuca** [community/community#128602] [Discussions]
### Select Topic Area

Product Feedback

### Body

Since many people would like to delete their posts (whether they are threads, pull requests or whatever) I recommend adding **Delete** button to the left of **Edit** one and one even in the menu that appears by clicking the 3 dots on the initial post. When the repository owner agrees the post is deleted shortly (to avoid deleting a post while someone is reading it a temporary notice of impending post deletion is shown) and when other users have interacted with it the request remains open for some time to give those users the time to choose. In addition, users will be able to decide whether receiving notifications for this feature or not. If not, their vote won't be necessary. It might be useful to allow users to discuss in a small side thread on the post page itself or loading it with a dedicated button.

Related discussions:
https://github.com/orgs/community/discussions/2948
https://github.com/orgs/community/discussions/46529
https://github.com/orgs/community/discussions/126059

**Comment by softking0503** [comment]
I think adding a "Delete" button next to the "Edit" button and in the dropdown menu (⋯) on posts, pull requests, or threads is a great idea. To ensure the process is smooth and avoids accidental deletions, here's a possible approach:

Deletion Request: When someone requests to delete a post, show a temporary notice that the post will be deleted shortly to inform anyone currently viewing it.

Pending State: If the post has interactions (e.g., comments), place the deletion request in a "Pending" state for a set period (like 24 hours) to give others time to react or save important content.

Notifications: Users can choose to receive notifications about deletion requests. By default, users who interacted with the post will be notified.

Discussion and Voting: Allow users to discuss the deletion in a small thread on the post page and vote on whether it should be deleted. The repository owner should have the final say.

Final Deletion: Once the request is approved or the pending period expires, the post is either deleted or the request is canceled.

This approach helps prevent accidental deletions and ensures that important posts are preserved while giving users a chance to participate in the decision-making process.

**Comment by LiteBrite82** [comment]
💬 Your Product Feedback Has Been Submitted 🎉
Thank you for taking the time to share your insights with us! Your feedback is invaluable as we build a better GitHub experience for all our users.

Here's what you can expect moving forward ⏩
- Your input will be carefully reviewed and cataloged by members of our product teams. 
- Other users may engage with your post, sharing their own perspectives or experiences. 
- GitHub staff may reach out for further clarification or insight. 

Where to look to see what's shipping 👀
- Read the [Changelog](https://github.blog/changelog/) for real-time updates on the latest GitHub features, enhancements, and calls for feedback.
- Explore our [Product Roadmap](https://github.com/orgs/github/projects/4247), which details upcoming major releases and initiatives.

What you can do in the meantime 💻
- Upvote and comment on other user feedback Discussions that resonate with you.
- Add more information at any point! Useful details include: use cases, relevant labels, desired outcomes, and any accompanying screenshots.

As a member of the GitHub community, your participation is essential. While we can't promise that every suggestion will be implemented, we want to emphasize that your feedback is instrumental in guiding our decisions and priorities.

Thank you once again for your contribution to making GitHub even better! We're grateful for your ongoing support and collaboration in shaping the future of our platform. ⭐
